clients/osv/src/main/kotlin/OsvService.kt

```diff
@@ -43,10 +43,12 @@ class OsvService(serverUrl: String = OsvApiClient.SERVER_URL_PRODUCTION, httpCli
         request: VulnerabilitiesForPackageRequest
     ): Result<List<Vulnerability>> {
         val response = client.getVulnerabilitiesForPackage(request).execute()
+        val body = response.body()
 
-        return when (response.isSuccessful) {
-            true -> Result.success(response.body()!!.vulnerabilities)
-            else -> Result.failure(IOException(response.message()))
+        return if (response.isSuccessful && body != null) {
+            Result.success(body.vulnerabilities)
+        } else {
```

Comment:
> if the result is success then the body must not be empty.

I believe instead of "empty" you mean "null", so I guess what you're trying to say is: Previously, a null body threw an exception due to the failing `!!` cast, and now a null-body is "silently" turned into a failure result, and the message of the wrapped `IOException` might not reveal the reason for the body to be null.

While that might indeed be, I believe the new behavior better fits the semantics of the function, because as a user I'd not expect a function that returns a `Result` to throw, but always wrap any exception into a `Result.failure`.

As a side note, it should anyway never happen that the body is null; we just need to jump through some hoops to please Kotlin's nullability checks, and [upcoming versions of OkHttp will make the body non-nullable](https://github.com/square/okhttp/blob/master/CHANGELOG.md#version-500-alpha7).

Reply:
>  I believe the new behavior better fits the semantics of the function, because as a user I'd not expect a function that returns a Result to throw

I believe the reasoning why it's better gets invalidated by your next sentence saying that

> As a side note, it should anyway never happen

In my view this change sacrifices on readability, and as it should never happen I prefer not to sacrifice readability.

So, I do not agree with that change but won't block it.